packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -717,6 +720,85 @@ class PhysicalModel extends SingleChildRenderObjectWidget {
   }
 }
 
+/// A widget representing a physical layer that clips its children to a shape.
+///
+/// Physical layers cast shadows based on an [elevation] which is nominally in
+/// logical pixels, coming vertically out of the rendering surface.
+///
+/// [PhysicalModel] does the same but only supports shapes that can be expressed
+/// as rectangles with rounded corners.
+class PhysicalShape extends SingleChildRenderObjectWidget {
+  /// Creates a physical model with an arbitrary shape clip.
+  ///
+  /// The [color] is required; physical things have a color.
+  ///
+  /// The [shape], [elevation], [color], and [shadowColor] must not be null.
+  const PhysicalShape({
+    Key key,
+    @required this.clipper,
+    this.elevation: 0.0,
+    @required this.color,
+    this.shadowColor: const Color(0xFF000000),
+    Widget child,
+    this.textDirection,
+  }) : assert(clipper != null),
+       assert(elevation != null),
+       assert(color != null),
+       assert(shadowColor != null),
+       super(key: key, child: child);
+
+  /// Determines which clip to use.
+  final CustomClipper<Path> clipper;
+
+  /// The z-coordinate at which to place this physical object.
+  final double elevation;
+
+  /// The background color.
+  final Color color;
+
+  /// The shadow color.
```

Comment:
this documentation does not satisfy our style guide

Reply:
I would just remove these comments, I don't find useful information to add in here.
I can make it "the color with which to fill the path", but isn't it just saying background color in more words?

But removing it will make the analyzer unhappy; so I resorted to just do the same as we did for the PhysicalModel widget.

Reply:
Please do prefer to omit the documentation entirely rather than including documentation that doesn't fit our style guide. We can figure out better documentation later. The analyzer won't complain (it'll just affect one of our benchmarks).

Reply:
FWIW, when writing documentation like this, I try to use the following prompts:
- If someone is looking at this documentation, it means that they have a question which they couldn't answer by guesswork or by looking at the code. What could that question be? Try to answer all questions you can come up with.
- If you were telling someone about this property, what might they want to know that they couldn't guess? For example, are there edge cases that aren't intuitive?
- Consider the type of the property or arguments. Are there cases that are outside the normal range that should be discussed? e.g. negative numbers, non-integer values, transparent colors, empty arrays, infinities, NaN, null? Discuss any that are non-trivial. For example, in this case, what happens if the shadow is opaque, or is transparent?
- Do this member interact with any others? For example, can it only be non-null if another is null? For example, in this case, `shadowColor` only matters if `elevation` is non-zero.
- Does this member have a similar name or purpose to another, such that we should point to that one, and from that one to this one? For example, in this case, `color` and `shadowColor` might be related and could point to each other.
- Are there timing considerations?
- Are there lifecycle considerations? For example, who owns the object that this property is set to? Who should dispose() it, if that's relevant?
- What is the contract for this property/method?

Reply:
done